google/cloud/internal/default_completion_queue_impl.cc

```diff
@@ -48,42 +48,45 @@ namespace {
  */
 class AsyncTimerFuture : public internal::AsyncGrpcOperation {
  public:
-  explicit AsyncTimerFuture(std::unique_ptr<grpc::Alarm> alarm)
-      : promise_(/*cancellation_callback=*/[this] { Cancel(); }),
-        alarm_(std::move(alarm)) {}
-
-  future<StatusOr<std::chrono::system_clock::time_point>> GetFuture() {
-    return promise_.get_future();
+  using ValueType = StatusOr<std::chrono::system_clock::time_point>;
+
+  // We need to create the shared_ptr before completing the initialization, so
+  // use a factory member function.
+  static std::pair<std::shared_ptr<AsyncTimerFuture>, future<ValueType>>
+  Create() {
+    auto self = std::shared_ptr<AsyncTimerFuture>(new AsyncTimerFuture);
```

Comment:
`std::make_shared()`?

Reply:
I do not think I can, the constructor is private (because the initialization is messy).